src/components/Upload.tsx

```diff
@@ -109,7 +109,20 @@ function Upload(): JSX.Element {
 					<Alert variant="primary" className="text-center py-3">
 						{state.selectedFile && (
 							<h5>
-								<b>Selected file:</b> {state.selectedFile.name}
+								<span title={state.selectedFile.name}>
+									<b>Selected file:</b>{" "}
+									{state.selectedFile.name.length > 17
+										? state.selectedFile.name.substring(
+												0,
+												17
+										  ) +
+										  " (...) " +
+										  state.selectedFile.name.substring(
+												state.selectedFile.name.length -
+													4
+										  )
+										: state.selectedFile.name}
+								</span>
```

Comment:
You can wrap this in a function like this
```ts
function getTruncatedFilename() {
   return <your logic>
}
```

then declare it as a variable before the return statement

then instead of something like
```ts
<b>Selected file:</b>{" "}
{filename}
```

you can do

```ts
<b>Selected file:</b>
{" " + filename}
```

Reply:
note: accidentally made this fix on main branch and then committed, so this does not need to be merged anymore